specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/ReplicationLinks.json

```diff
@@ -436,14 +552,48 @@
           "readOnly": true
         },
         "linkType": {
-          "description": "Link type (GEO, NAMED, STANDBY).",
+          "description": "Link type (GEO, NAMED, STANDBY). Update operation does not support NAMED.",
+          "enum": [
+            "GEO",
+            "NAMED",
+            "STANDBY"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ReplicationLinkType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "ReplicationLinkUpdate": {
+      "description": "A replication link update request.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ReplicationLinkUpdateProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ReplicationLinkUpdateProperties": {
+      "description": "Properties of a replication link update.",
+      "type": "object",
+      "properties": {
+        "linkType": {
+          "description": "Link type (GEO, NAMED, STANDBY). Update operation does not support NAMED.",
```

Comment:
Then, why is NAMED being added to linkType enum in Update Request Payload?

Reply:
It is using the same link type enum as the other controllers. Just to make it consistent with others and eventually in the docs, same description has been used.

Reply:
Why dont you create a new definition then ? THere is no point in making things consistent if they are not actually work the same way .

Reply:
we want to keep PUT and the PATCH controller consistent with each other. 
PUT allows for a GET/PUT cycle, so if customer takes the body from a GET request, they should be allowed to PUT the same content back. Creating a new definition/parameter for the same entity would just add to confusion of the Customer and implementation overhead.
Also, we might add the support for NAMED in future.

Reply:
The object in question is being PATCHed not PUT. the GET\PUT cycle has nothing to do with this. 

in the PATCH operation by disallowing this in the type , you directly enable to client to prevent sending the "NAMED" parameter at all. This is more helpful to the customer and there is no confusion here since the customer is not interacting with these individual type definitions.

Reply:
The PATCH behavior on this property is the same as PUT, we are designing this API for the future where we will be supporting NAMED, just not at this time. If it makes it clearer, we can remove the "Update operation does not support NAMED" if that makes more sense.

Reply:
The object in question is being PATCHED, but we are supporting the same PATCH functionality through PUT as well (in order to make the ARM deployment templates work, because they seem to support only PUT)
But here, PUT does not support create or update of any other parameter, it just supports the update link type (same as PATCH).

The resource property class (defining the entity which is used for PUT request body) already has a definition for link type, so it makes more sense to utilize the existing parameter for PUT to avoid redundancy and confusion. Also, keeping in mind that we might add support for NAMED in future.

Creating a new definition for PATCH request payload will cause inconsistency and confusion between PUT and PATCH.

Also, I see such cases in other APIs where only the documentation has been modified.
[Databases - Create Or Update - REST API (Azure SQL Database) | Microsoft Learn](https://learn.microsoft.com/en-us/rest/api/sql/2021-11-01/databases/create-or-update?tabs=HTTP#sku) this page says "Copy, Secondary, and RestoreLongTermRetentionBackup are not supported for DataWarehouse edition" for CreateMode parameter.

Reply:
> The PATCH behavior on this property is the same as PUT, we are designing this API for the future where we will be supporting NAMED, just not at this time. If it makes it clearer, we can remove the "Update operation does not support NAMED" if that makes more sense.

we could do this as well, makes more sense

Reply:
As discussed in the internal chat, we are fine with the current approach.